### ISSUE TITLE
Not allowing custom nav types

### ISSUE THREAD

**Issue by pablobaldez** [stefanoq21/BottomSheetNavigator3#4]
The `NavGraphBuilder.bottomSheet` function is now allowing custom navTypes, which is a requirement if the route class has parameters. It uses a empty map instead
```kotlin
inline fun <reified T : Any> NavGraphBuilder.bottomSheet(
    deepLinks: List<NavDeepLink> = emptyList(),
    noinline content: @Composable ColumnScope.(backstackEntry: NavBackStackEntry) -> Unit
) {
    destination(
        BottomSheetNavigatorDestinationBuilder(
            provider[BottomSheetNavigator::class],
            T::class,
            emptyMap(),
            content
        )
            .apply {
                deepLinks.forEach { deepLink -> deepLink(deepLink) }
            }
    )
}
```

My suggestion is to add a map `Map<KType, @JvmSuppressWildcards NavType<*>>` on its parameter, allowing the transference between custom types, such as Enums or Parcelables.
The standaard `composable` has it in the type-safe implementation.

(I tried to open a PR but I think the repo is not opened for writing contributions)

**Comment by stefanoq21** [comment]
Hi @pablobaldez I will check this as soon as possible. Thanks.

**Comment by stefanoq21** [comment]
Hi @pablobaldez, thanks for the suggestion. It's included in version 0.0.6.
